bindings/pyroot_experimental/PyROOT/test/tdirectory.py

```diff
@@ -0,0 +1,65 @@
+import os
+import unittest
+
+import ROOT
+from libcppyy import SetOwnership
+
+
+class TDirectoryUnits(unittest.TestCase):
+    """
+    Test for the pythonizations of TFile
+    """
+
+    dirName  = 'myDir'
+    dirName2  = 'myDir2'
+    directory = None
+    histName  = 'myHist'
+    keyName = 'myKey'
+    fileName  = 'tdirectoryWrite.root'
+    hist = None
+    nbins = 128
+    minx = -4
+    maxx = 4
+
+    # Setup
+    @classmethod
+    def setUpClass(cls):
+       cls.directory = ROOT.gDirectory.mkdir(cls.dirName)
+       cls.hist = ROOT.TH1F(cls.histName, cls.histName, cls.nbins, cls.minx, cls.maxx)
+       cls.hist.SetDirectory(cls.directory)
+       SetOwnership(cls.hist, False) # Is this a bug in cppyy?
```

Comment:
It is not a bug, it is just Cppyy not being aware of the internal memory management of ROOT. In the current PyROOT, there is some code that takes care of this, but such code has not been ported to experimental PyROOT yet.

The topic of how the memory management should be in the future PyROOT actually needs a longer discussion that already started at the ROOT workshop.

Reply:
I fully agree. Technically this is a feature. We really need to fix this for cppyy... I think that once this is done, it will become already usable.